pkg/backup/s3/s3.go

```diff
@@ -30,37 +27,42 @@ const (
 	v1 = "v1/"
 )
 
-// S3 represents AWS S3 service.
+// S3 is to translate operator's S3 storage logic (e.g. dir layout) into S3 native code.
 type S3 struct {
 	bucket string
 	prefix string
 	client *s3.S3
 }
 
+// New returns a S3 translator from default shared config.
 // Please refer to http://docs.aws.amazon.com/sdk-for-go/api/aws/session/
-// for how to set credentials and configuration when creating a session.
-func New(bucket, prefix string, option session.Options) (*S3, error) {
-	if bucket == "" {
-		return nil, fmt.Errorf("env (%s) must be set", env.AWSS3Bucket)
-	}
-	sess, err := session.NewSessionWithOptions(option)
+// for setting up credentials and configuration.
+func New(bucket, namespace, clusterName string) (*S3, error) {
```

Comment:
i feel this is a step backward. why do we need to introduce k8s/etcd specific concept (namespace, clustername) into backup/s3 when we can avoid them?